### PR TITLE
Unit test for nnimanager

### PR DESCRIPTION
**Pull request by demianzhang**
Unit test for nnimanager. Test the public function without resumeExperiment( ).

### CODE REVIEW

**Comment by chicm-ms** [comment]
Please remove --exclude core/test/nnimanager.test.ts  from package.json

File: src/nni_manager/core/test/nnimanager.test.ts [hunk 1]
```diff
@@ -71,29 +73,71 @@ describe('Unit test for nnimanager', function () {
...
-        //return nniManager.listTrialJobs().then(function (trialJobDetails) {
-        //    expect(trialJobDetails.length).to.be.equal(2);
-        //}).catch(function (error) {
-        //    assert.fail(error);
-        //})
+        return nniManager.listTrialJobs().then(function (trialjobdetails) {
+            expect(trialjobdetails.length).to.be.equal(2);
+            expect(trialjobdetails[0].id).to.be.equal('4321'); // generated by init
+            expect(trialjobdetails[0].status).to.be.equal('WAITING');
+            expect(trialjobdetails[1].id).to.be.equal(''); // generated by addCustomizedTrialJob
```
Comment (chicm-ms) [inline src/nni_manager/core/test/nnimanager.test.ts]: listTrialJobs methods does not guarantee order of jobs,  the first expect is enough for nnimanager UT.  The remaining 4 expects can be removed, they should be tested by datastore UT.

File: src/nni_manager/core/test/nnimanager.test.ts [hunk 2]
```diff
@@ -141,11 +184,106 @@ describe('Unit test for nnimanager', function () {
...
         }).catch((error) => {
             assert.fail(error);
         })
     })
+
+    it('test updateExperimentProfile TRIAL_CONCURRENCY',  () => {
+        return nniManager.updateExperimentProfile(experimentProfile, 'TRIAL_CONCURRENCY').then(() => {
+            nniManager.getExperimentProfile().then((updateProfile) => {
+                expect(updateProfile.params.trialConcurrency).to.be.equal(3)
+                });
```
Comment (chicm-ms) [inline src/nni_manager/core/test/nnimanager.test.ts]: indent seems incorrect here.

File: src/nni_manager/core/test/nnimanager.test.ts [hunk 3]
```diff
@@ -141,11 +184,106 @@ describe('Unit test for nnimanager', function () {
...
 
         }).catch((error) => {
             assert.fail(error);
         })
     })
+
+    it('test updateExperimentProfile TRIAL_CONCURRENCY',  () => {
+        return nniManager.updateExperimentProfile(experimentProfile, 'TRIAL_CONCURRENCY').then(() => {
+            nniManager.getExperimentProfile().then((updateProfile) => {
+                expect(updateProfile.params.trialConcurrency).to.be.equal(3)
```
Comment (chicm-ms) [inline src/nni_manager/core/test/nnimanager.test.ts]: missing ';'

File: src/nni_manager/core/test/nnimanager.test.ts [hunk 4]
```diff
@@ -71,29 +73,71 @@ describe('Unit test for nnimanager', function () {
...
-        //}).catch(function (error) {
-        //    assert.fail(error);
-        //})
+        return nniManager.listTrialJobs().then(function (trialjobdetails) {
+            expect(trialjobdetails.length).to.be.equal(2);
+            expect(trialjobdetails[0].id).to.be.equal('4321'); // generated by init
+            expect(trialjobdetails[0].status).to.be.equal('WAITING');
+            expect(trialjobdetails[1].id).to.be.equal(''); // generated by addCustomizedTrialJob
+            expect(trialjobdetails[1].status).to.be.equal('WAITING');
+        }).catch(function (error) {
```
Comment (chicm-ms) [inline src/nni_manager/core/test/nnimanager.test.ts]: Please use arrow function here.

File: src/nni_manager/core/test/nnimanager.test.ts [hunk 5]
```diff
@@ -141,11 +184,106 @@ describe('Unit test for nnimanager', function () {
...
+        })
+    })
+
+    it('test getTrialJobStatistics', () => {
+        // get 3 trial jobs (init, addCustomizedTrialJob, cancelTrialJobByUser)
+        return nniManager.getTrialJobStatistics().then(function (trialJobStatistics) {
+            expect(trialJobStatistics.length).to.be.equal(2);
+            expect(trialJobStatistics[0].trialJobStatus).to.be.equal('WAITING');
+            expect(trialJobStatistics[0].trialJobNumber).to.be.equal(2);
+            expect(trialJobStatistics[1].trialJobStatus).to.be.equal('USER_CANCELED');
```
Comment (chicm-ms) [inline src/nni_manager/core/test/nnimanager.test.ts]: Here we should not assume 'WAITING' is in trialJobStatistics[0],  please use another way to check.

File: src/nni_manager/core/test/nnimanager.test.ts [hunk 6]
```diff
@@ -71,36 +73,74 @@ describe('Unit test for nnimanager', function () {
...
+        await setTimeout(() => {nniManager.stopExperiment()},15000);
         cleanupUnitTest();
     })
 
-    it('test resumeExperiment', () => {
-        //TODO: add resume experiment unit test
+
+
+    it('test addCustomizedTrialJob', () => {
+        return nniManager.addCustomizedTrialJob('hyperParams').then(() => {
```
Comment (QuanluZhang) [inline src/nni_manager/core/test/nnimanager.test.ts]: passing 'hyperParams' does not make sense...

**Pull request merged**